src/sentry/profiles/flamegraph.py

```diff
@@ -142,51 +141,66 @@ def get_profile_ids_with_spans(
     project_id: int,
     params: ParamsType,
     span_group: str,
-    backend: str,
-    query: Optional[str] = None,
 ):
-    data = query_profiles_data(
-        params,
-        Referrer.API_STARFISH_PROFILE_FLAMEGRAPH.value,
-        selected_columns=["id", "profile.id"],
-        query=query,
-        additional_conditions=[
-            Condition(Function("has", [Column("spans.group"), span_group]), Op.EQ, 1)
+    query = Query(
+        match=Entity(EntityKey.Spans.value),
+        select=[
+            Column("profile_id"),
+            Function(
+                "groupArray",
```

Comment:
I'm not familiar with this dataset but you may want to impose a limit here as a safeguard in case this array grows too large and OOMs in clickhouse

```suggestion
                "groupArray(10)",
```

Reply:
@Zylphrex basically this is an array of span intervals (_start_ and _end_) within a transaction (with a profile). So it can get as big as the number of spans within a single transaction.

While I think it'd be nice to have a safeguard I think the issue with respect to a generic LIMIT clause is that here we'd be cutting/removing random span intervals from a given transaction and maybe omitting an interesting part of a profile for flamegraph aggregation.

If we fear that the number of spans within a transaction could be so big to cause OOMs, do you know what would be a good threshold to stay below (snuba-wise)? I don't have stats on the avg. number of spans within transactions but if we have some margin to increase it further I'd rather have something bigger than `10`.

Reply:
Historically, we were limited to 1000 spans per transaction. So you could pick something like 100 and likely be good enough for most cases.